artifactory/utils/buildinfoproperties.go

```diff
@@ -226,18 +198,26 @@ func CreateBuildInfoPropertiesFile(buildName, buildNumber, projectKey, deployabl
 	// If a value is provided by the build config file write it,
 	// otherwise use the default value from defaultPropertiesValues map.
 	for _, partialMapping := range buildTypeConfigMapping[projectType] {
-		for propertyKey, configKey := range *partialMapping {
+		for propKey, configKey := range *partialMapping {
 			if config.IsSet(configKey) {
-				_, err = propertiesFile.WriteString(propertyKey + "=" + config.GetString(configKey) + "\n")
-			} else if defaultVal, ok := defaultPropertiesValues[propertyKey]; ok {
-				_, err = propertiesFile.WriteString(propertyKey + "=" + defaultVal + "\n")
-			}
-			if err != nil {
-				return "", errorutils.CheckError(err)
+				props[propKey] = config.GetString(configKey)
+				addIncubatingProps(props, propKey, config.GetString(configKey))
+			} else if defaultVal, ok := defaultPropertiesValues[propKey]; ok {
+				props[propKey] = defaultVal
+				addIncubatingProps(props, propKey, defaultVal)
+
 			}
 		}
 	}
-	return propertiesFile.Name(), nil
+	return props, nil
+}
+
+// Properties that have the 'artifactory.' prefix are deprecated.
+// For backward compatibility reasons, both will be added to the props map.
+func addIncubatingProps(props map[string]string, propKey, value string) {
```

Comment:
Let's change this function's name to ```addProp```, and let it do everything - 

1. Remove the prefix if needed,.
2. Add the prop to the map.

I think this will simplify things/